docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+decisions.
+
+### 4.1. Mode Switch Feature
+
+#### 4.1.1. Implementation
+
+The mode switch mechanism is facilitated by `AppState`. It contains an `AppMode` object and can be accessed from
+`Mode Switch` object.
+
+The following sequence diagram shows how the mode switch operation works:
+
+![Sequence Diagram for Mode Switch Command](diagrams/class/jpeg/SequenceDiagram_ModeSwitch.jpg)
+
+Given below is an example usage scenario and how the find mode switch function behaves.
+
+Step 1. The user launches the application for the first time. The `AppState` object is constructed and the `AppMode`
+field is set to `TIMETABLE` by default.
+
+Step 2. The user types `mode /n`. The `mode /t` command is passed through
+`InputParser#getCommandFromInput`, which constructs a `ModeSwitch` object and calls `ModeSwitch#execute()`.
+
+Step 3. `execute()` is called, which then set the `AppMode` field in the `AppState` object either to `TIMETABLE`
+or `NOTEBOOK_SHELF` or throw an `InvalidCommandException`.
+
+Step 4. To signal that the user has successfully changed the mode, a message is printed with the current mode of the
+program.  
+
+#### 4.1.2. Design Considerations
+
+/* work in progress */
+
+### 4.2. Timetable Mode
+
+#### 4.2.1. TaskList Management Feature
+
+##### 4.2.1.1. Implementation
+
+`TaskList` is implemented to manage and store the tasks input by the user. It comprises an `ArrayList` list of
+ `Task`s, and a few helper methods.
+
+This means that multiple operations such as addition and deletion can be done on a `Task`, without affecting
+the contents of other `Task` in the `TaskList`.
+<br>
+
+Given below is an example usage scenario and how the add task function behaves.
+
+1. The user launches the application for the first time. CliUserInterface#executeCommand is called when the user adds a `Task` into the `TaskList`.
+
+2. The user types `add /tTask /by19-10-2020 1900`. The `add` command is passed through `InputParser#getCommandFromInput`, which then creates a constructor for `AddCommandTimetableMode`.
+
+3. `AddCommandTimetableMode#execute()` is called, which then calls `InputParser#parseTaskTitle`, which first extracts the `title` from the user's input.
+
+4. `InputParser#parseDeadline` is then called, which returns the `deadline` to `AddCommandTimetableMode#execute()`.
+
+5. `TaskList#addTask` is then called and a new `Task`, with `title` and `deadline`, is initialised.
+
+6. To signal that the user has successfully added a task, a message is printed with `CliMessages#printAddedTaskMessage`.
+
+The UML sequence diagram below shows how the add task command works.
+![Sequence Diagram for Add Task Command](diagrams/class/jpeg/add_task.jpg)
+
+<hr>
+TaskList also allows the deletion of tasks by the user.
+
+The figure below shows how the delete task command works:
+![Sequence Diagram for Delete Task Command](diagrams/class/jpeg/SequenceDiagram_RemoveTimetable.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/12445358/97579783-709b2b80-1a2d-11eb-8b09-c7ff10446bb4.png)
Please try to fix the aspect ratio as it is hard to read the diagram.